internal/bootloader/grub.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package bootloader
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/canonical/pebble/internal/bootloader/grubenv"
+	"github.com/canonical/pebble/internal/osutil"
+)
+
+// GRUB implements the Bootloader interface to support bootloader operations
+// on GRUB 2.
+type GRUB struct {
+	// rootdir is the directory where the GRUB prefix is mounted.
+	rootdir string
+}
+
+// newGrub initializes a new instance of the GRUB bootloader.
+func NewGRUB(rootdir string) Bootloader {
```

Comment:
These don't need to be exported anymore. However, we do need to return a `Bootloader` because we need these functions to fit the `func(rootdir string) Bootloader` mold.